performance/api_load_test.go

```diff
@@ -0,0 +1,212 @@
+package performance
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/internal/cltest"
+	"github.com/stretchr/testify/assert"
+	"github.com/tsenart/vegeta/lib"
+)
+
+type Requirement struct {
+	requestsPerSecond int
+	averageLatencyMax time.Duration
+	percentileMax     time.Duration
+	runDurationMax    time.Duration
+}
+
+var (
+	r            Requirement
+	requirements = []Requirement{
+		{20, time.Millisecond * 100, time.Millisecond * 100, time.Millisecond * 100},
+		{40, time.Millisecond * 200, time.Millisecond * 200, time.Millisecond * 200},
+		{60, time.Millisecond * 300, time.Millisecond * 300, time.Millisecond * 300},
+	}
+
+	app     *cltest.TestApplication
+	tests   func(t *testing.T)
+	cleanup func()
+)
+
+func Setup() {
+	tests = func(t *testing.T) {
+		for _, requirement := range requirements {
+			app, cleanup = cltest.NewApplicationWithKeyStore()
+			r = requirement
+			t.Log(fmt.Printf("Requests per second: %d\n", r.requestsPerSecond))
+			t.Run("CreateJobs", CreateJobApiUnderLoad)
+			t.Run("ViewJobs", ViewJobApiUnderLoad)
+			t.Run("CreateJobRuns", CreateJobRunsApiUnderLoad)
+			t.Run("ViewJobRuns", ViewJobRunsApiUnderLoad)
+			cleanup()
+		}
+	}
+	return
+}
+
+func TestNodeUnderLoad(t *testing.T) {
+	Setup()
+	tests(t)
```

Comment:
In general, we lean towards share nothing architectures. In common terms, that means that you don't share global `var`s and always pass and return copies.

The flow I see here is:
1. Define `tests` as type func(...)
2. Assign tests in `Setup`
3. Invoke `tests(t)`

Why not just have a method return `tests` and then invoke it, avoiding the global?